pkg/hubble/relay/pool/option.go

```diff
@@ -42,7 +42,8 @@ var DefaultOptions = Options{
 		Max:    12 * time.Hour,
 		Factor: 2.0,
 	},
-	RetryTimeout: 30 * time.Second,
+	ConnCheckInterval: 2 * time.Minute,
```

Comment:
> I think if we only trigger reconnects every `ConnCheckInterval`

Reconnects are also triggered with a client requests that uses `ReportOffline`.

> The first retry which is supposed to happen after 10 seconds according to exponential backoff will actually happen after 2 minutes.

Unless there's a client request, yes, this is true. I'm not sure it's worth trying to reconnect more aggressively than this.

> In addition, we will start all re-connection attempts always at the same time for all offline peers whose backoff has expired in the last 2 minutes. There is no jitter.

Very good point. I thought about this as well but as this PR was already getting pretty big, I thought this could be addressed in a follow-up PR. Note that even before this PR there's also anbother burst when Hubble Relay is started and is notified of all the peers in the cluster.

I fully agree with your point here and I'm happy to discuss a solution to implement in a follow-up PR.

Reply:
> Unless there's a client request, yes, this is true. I'm not sure it's worth trying to reconnect more aggressively than this.

If there is a client reconnect, then the backoff is ignored anwyway. Currently, actual back-off function will look like this. 

 ```
2m0s (ConnCheckInterval)
2m0s (ConnCheckInterval)
2m0s (ConnCheckInterval)
2m0s (ConnCheckInterval)
2m40s
5m20s
10m40s
21m20s
42m40s
1h25m20s
2h50m40s
5h41m20s
11h22m40s
12h0m0s
```

I agree that we do not have to fix this in this PR, but it seems like this is not intentional.